packages/ai-jsx/src/batteries/constrained-output.tsx

```diff
@@ -244,8 +244,12 @@ async function* ObjectCompletionWithRetry(
     output = yield* render(childrenWithCompletion);
     return output;
   } catch (e: any) {
-    validationError = e.metadata.validationError;
-    output = e.metadata.output;
+    if (e instanceof CompletionError) {
+      validationError = e.metadata.validationError;
+      output = e.metadata.output;
+    } else {
+      throw e;
```

Comment:
Nice. It feels safer to only explicitly handle errors we know about and throw the rest.